src/renderer/components/ft-playlist-add-video-button/ft-playlist-add-video-button.js

```diff
@@ -0,0 +1,135 @@
+import { defineComponent } from 'vue'
+import FtIconButton from '../ft-icon-button/ft-icon-button.vue'
+import { mapActions } from 'vuex'
+
+export default defineComponent({
+  name: 'FtPlaylistAddVideoButton',
+  components: {
+    'ft-icon-button': FtIconButton,
+  },
+  props: {
+    videoId: {
+      type: String,
+      required: true
+    },
+    videoTitle: {
+      type: String,
+      required: true
+    },
+    channelId: {
+      type: String,
+      required: true
+    },
+    channelName: {
+      type: String,
+      required: true
+    },
+    viewCount: {
+      type: Number,
+      required: false,
+      default: null,
+    },
+    lengthSeconds: {
+      type: Number,
+      required: true
+    },
+
+    padding: {
+      type: Number,
+      default: 10
+    },
+    size: {
+      type: Number,
+      default: 20
+    },
+
+    showVideoAddedToPlaylistCount: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  computed: {
+    allPlaylists: function () {
+      return this.$store.getters.getAllPlaylists
+    },
+
+    title() {
+      const initialText = this.$t('User Playlists.Add to Playlist')
+      if (this.videoAddedToPlaylistCountText == null) { return initialText }
+
+      return `${initialText} (${this.videoAddedToPlaylistCountText})`
+    },
+
+    videoAddedToPlaylistCount() {
+      if (!this.showVideoAddedToPlaylistCount) { return 0 }
+
+      // Accessing a reactive property has a negligible amount of overhead,
+      // however as we know that some users have playlists that have more than 10k items in them
+      // it adds up quickly, especially as there are usually lots of ft-list-video instances active at the same time.
+      // So create a temporary variable outside of the array, so we only have to do it once.
+      // Also the search is re-triggered every time any playlist is modified.
+      const videoId = this.videoId
+      let count = 0
+
+      this.allPlaylists.forEach((playlist) => {
+        const videoAlreadyAdded = playlist.videos.some((v) => {
+          return v.videoId === videoId
+        })
+        if (videoAlreadyAdded) { count += 1 }
+      })
+
+      // If only saved in quick bookmark target, don't show the count which is confusing
+      if (count === 1 && this.isInQuickBookmarkPlaylist) {
+        return 0
+      }
```

Comment:
We don't need to check the quick bookmark playlist again, as we already know if it contains the video thanks to the `isInQuickBookmarkPlaylist` prop.
```suggestion
      const isQuickBookmarkEnabled = this.isQuickBookmarkEnabled
      let quickBookmarkPlaylistId

      if (isQuickBookmarkEnabled) {
        quickBookmarkPlaylistId = this.quickBookmarkPlaylist._id
      }

      this.allPlaylists.forEach((playlist) => {
        // We don't need to check the quick bookmark playlist
        // as we already know it from isInQuickBookmarkPlaylist
        if (isQuickBookmarkEnabled && quickBookmarkPlaylistId === playlist._id) {
          return
        }

        const videoAlreadyAdded = playlist.videos.some((v) => {
          return v.videoId === videoId
        })
        if (videoAlreadyAdded) { count++ }
      })

      // If only saved in quick bookmark target, don't show the count which is confusing
      if (count > 0 && this.isInQuickBookmarkPlaylist) {
        count++
      }
```